tests/test_iib_client.py

```diff
@@ -117,31 +127,59 @@ def fixture_builds_page2_json(fixture_build_details_json2):
     return json
 
 
-def test_iib_client(fixture_build_details_json, fixture_builds_page1_json):
+def test_iib_client(
+    fixture_add_build_details_json,
+    fixture_rm_build_details_json,
+    fixture_regenerate_bundle_build_details_json,
+    fixture_builds_page1_json,
+):
     with requests_mock.Mocker() as m:
         m.register_uri(
             "POST",
             "/api/v1/builds/add",
             status_code=200,
-            json=fixture_build_details_json,
+            json=fixture_add_build_details_json,
         )
         m.register_uri(
             "POST",
             "/api/v1/builds/rm",
             status_code=200,
-            json=fixture_build_details_json,
+            json=fixture_rm_build_details_json,
+        )
+        # TODO add other request types
+
+        m.register_uri(
+            "POST",
+            "/api/v1/builds/regenerate-bundle",
+            status_code=200,
+            json=fixture_regenerate_bundle_build_details_json,
         )
         m.register_uri(
             "GET", "/api/v1/builds", status_code=200, json=fixture_builds_page1_json
         )
         m.register_uri(
-            "GET", "/api/v1/builds/1", status_code=200, json=fixture_build_details_json
+            "GET",
+            "/api/v1/builds/1",
+            status_code=200,
+            json=fixture_add_build_details_json,
+        )
+        m.register_uri(
+            "GET",
+            "/api/v1/builds/2",
+            status_code=200,
+            json=fixture_rm_build_details_json,
+        )
+        m.register_uri(
+            "GET",
+            "/api/v1/builds/3",
+            status_code=200,
+            json=fixture_regenerate_bundle_build_details_json,
         )
 
         iibc = IIBClient("fake-host")
         assert iibc.add_bundles(
             "index-image", ["bundles-map"], [], binary_image="binary"
-        ) == IIBBuildDetailsModel.from_dict(fixture_build_details_json)
+        ) == IIBBuildDetailsModel.from_dict(fixture_add_build_details_json)
```

Comment:
Do you think that it would be good idea to use actual class for creating the object? So people see that this is better approach because it also checks the data (request_type).

`AddModel.from_dict(fixture_add_build_details_json)`

Tip: can be replaced everywhere easily (e.g. find/replace in PyCharm). We would want to keep it (test it) just in from_dict test.

Reply:
A good idea! I changed the classes where it made sence.